### PR TITLE
feat: multi-arch docker images

### PR DESCRIPTION
**Pull request by thiskevinwang**
## "Ready for Review" checklist

- [ ] The Vercel preview link has been added to this PR's description
- [ ] The Asana task has been added to this PR's description
- [ ] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [ ] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [ ] The description template has been filled in below

---

## Relevant links

- [Preview link](url) 🔎
- [Asana task](url) 🎟️

### CODE REVIEW

File: .github/workflows/docker_publish.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ name: Docker Publish
 # documentation.
 
 on:
+  pull_request:
```
Comment (thiskevinwang) [inline .github/workflows/docker_publish.yml:9]: 💭 Do we want to verify that Docker builds properly on a PR? (no push occurs)

Comment (BRKalow) [reply .github/workflows/docker_publish.yml:9]: I don't think it's necessary, as all it really does is run npm install, but I don't feel too strongly about it. If we do want to keep it, we'll need to update something with the build and push step as it looks like the latest run pushed to the latest tag.

Comment (thiskevinwang) [reply .github/workflows/docker_publish.yml:9]: Will remove this!

Comment (thiskevinwang) [inline .github/workflows/docker_publish.yml:9]:
```suggestion
```

**Comment by BRKalow** [comment]
@thiskevinwang Do we need to remove [this](https://github.com/hashicorp/dev-portal/blob/main/Dockerfile#L1) flag from the Dockerfile, or do you know if it'll get overwritten by the platform specified from our config here?

**Comment by thiskevinwang** [comment]
> @thiskevinwang Do we need to remove [this](https://github.com/hashicorp/dev-portal/blob/main/Dockerfile#L1) flag from the Dockerfile, or do you know if it'll get overwritten by the platform specified from our config here?

@BRKalow ah good catch, also TIL 👇! ... Yes we should (I will) remove that

> All FROM commands inside the Dockerfile without their own --platform flag will pull base images for this platform and this value will also be the platform of the resulting image.
> - [docs](https://docs.docker.com/engine/reference/commandline/buildx_build/#platform)

**Review by BRKalow — APPROVED** [review]

**Pull request merged**